cpp/include/raft/neighbors/detail/cagra/cagra_search.cuh

```diff
@@ -118,7 +118,7 @@ void search_main(raft::resources const& res,
   RAFT_EXPECTS(queries.extent(1) == index.dim(), "Queries and index dim must match");
   const uint32_t topk = neighbors.extent(1);
 
-  if (params.max_queries == 0) { params.max_queries = queries.extent(0); }
+  if (params.max_queries == 0) { params.max_queries = std::min<size_t>(queries.extent(0), 65535); }
```

Comment:
It would be more robust to compute the value, perhaps using std::limits for int 16, MAX_UNSIGNED_SHORT, or (1<<16)-1.